### PR TITLE
Add createTransaction.js test

### PR DESCRIPTION
**Pull request by ilake**

### CODE REVIEW

File: test/createTransaction.js [hunk 1]
```diff
@@ -1,35 +1,132 @@
...
+      assert.equal(eventArgs.metadata, metadata)
+    })
+
+    it("transfers buyer's tokens to escrow", async () => {
+      let xfer = await protocol.transfer(buyer, amount)
+      let eventArgs = $util.eventFromTx(xfer, $util.events.Transfer).args
+
+      // verify transfer results
+      assert.equal(eventArgs.to, buyer)
+      assert.equal(eventArgs.value, amount)
```
Comment (ngan) [inline test/createTransaction.js]: We don't need to test XNK transfer to the buyer.

File: test/createTransaction.js [hunk 2]
```diff
@@ -1,35 +1,132 @@
...
+
+      assert.equal(await $util.getBalance(protocol.address, protocol), amount)
+    })
 
     describe("when mediator is specified", () => {
-      it("fails when policy is not specified")
-      it("fails when mediator rejects the transaction")
-      it("passes the transaction's id, amount, and owner the mediator")
+      it("fails when policy is not specified", async () => {
+        let policy_address = 0;
```
Comment (ngan) [inline test/createTransaction.js]: Please use camel casing.

**Review by ngan — CHANGES_REQUESTED** [review]

File: test/createTransaction.js [hunk 3]
```diff
@@ -1,35 +1,154 @@
...
+      eventArgs0 = $util.eventFromTx(tx0, $util.events.TransactionInitiated).args
+      eventArgs1 = $util.eventFromTx(tx1, $util.events.TransactionInitiated).args
+      assert.equal(eventArgs0.id.toNumber(), 0)
+      assert.equal(eventArgs1.id.toNumber(), 1)
+    })
+
+    it("emits the TransactionInitiated event", async () => {
+      let xfer = await protocol.transfer(buyer, amount)
+      let tx = await protocol.createTransaction(seller, amount, metadata, policy.address, mediator.address, 0, { from: buyer })
+      eventArgs = $util.eventFromTx(tx, $util.events.TransactionInitiated).args
```
Comment (ngan) [inline test/createTransaction.js]: Missing let

File: test/createTransaction.js [hunk 4]
```diff
@@ -1,35 +1,154 @@
...
+      await protocol.transfer(buyer, amount)
+      await $util.assertVMExceptionAsync(protocol.createTransaction(seller, amount, metadata, 0, mediator.address, 0, { from: buyer }))
+    })
+
+    it("increments the global transaction ID for the next transaction", async () => {
+      let xfer = await protocol.transfer(buyer, amount * 2)
+      let tx0 = await protocol.createTransaction(seller, amount, metadata, policy.address, mediator.address, 0, { from: buyer })
+      let tx1 = await protocol.createTransaction(seller, amount, metadata, policy.address, mediator.address, 0, { from: buyer })
+      eventArgs0 = $util.eventFromTx(tx0, $util.events.TransactionInitiated).args
+      eventArgs1 = $util.eventFromTx(tx1, $util.events.TransactionInitiated).args
```
Comment (ngan) [inline test/createTransaction.js]: Missing let

File: test/createTransaction.js [hunk 5]
```diff
@@ -1,35 +1,154 @@
...
+
+    it("transfers buyer's tokens to escrow", async () => {
+      let xfer = await protocol.transfer(buyer, amount)
+      let eventArgs = $util.eventFromTx(xfer, $util.events.Transfer).args
+
+      let tx = await protocol.createTransaction(seller, amount, metadata, policy.address, mediator.address, 0, { from: buyer })
+
+      eventArgs = $util.eventFromTx(tx, $util.events.TransactionInitiated).args
+      assert.equal(eventArgs.buyer, buyer)
+      assert.equal(eventArgs.seller, seller)
```
Comment (ngan) [inline test/createTransaction.js]: Why test for this?

**Review by ngan — CHANGES_REQUESTED** [review]

**Comment by ilake** [comment]
@ngan please review this, thanks

**Pull request merged**